src/mono/mono/metadata/custom-attrs.c

```diff
@@ -574,34 +580,44 @@ MONO_RESTORE_WARNING
 				if (!bcheck_blob (p, 7, boundp, error))
 					return NULL;
 				guint64 val = read64 (p);
-				mono_array_set_internal (arr, guint64, i, val);
+				if (arr)
+					mono_array_set_internal (arr, guint64, i, val);
 				p += 8;
 			}
 			break;
 		case MONO_TYPE_CLASS:
 		case MONO_TYPE_OBJECT:
 		case MONO_TYPE_STRING:
 		case MONO_TYPE_SZARRAY: {
-			HANDLE_FUNCTION_ENTER ();
-			MONO_HANDLE_NEW (MonoArray, arr);
-
-			for (i = 0; i < alen; i++) {
-				MonoObject *item = NULL;
-				load_cattr_value (image, m_class_get_byval_arg (tklass), &item, p, boundp, &p, error);
-				if (!is_ok (error))
-					return NULL;
-				mono_array_setref_internal (arr, i, item);
+			if (arr) {
+				HANDLE_FUNCTION_ENTER ();
+				MONO_HANDLE_NEW (MonoArray, arr);
+
+				for (i = 0; i < alen; i++) {
+					MonoObject *item = NULL;
+					load_cattr_value (image, m_class_get_byval_arg (tklass), &item, p, boundp, &p, error);
+					if (!is_ok (error))
+						return NULL;
+					mono_array_setref_internal (arr, i, item);
+				}
+				HANDLE_FUNCTION_RETURN ();
+			} else {
+				for (i = 0; i < alen; i++) {
+					MonoObject *item = NULL;
+					load_cattr_value (image, m_class_get_byval_arg (tklass), &item, p, boundp, &p, error);
+					if (!is_ok (error))
+						return NULL;
+				}
```

Comment:
This doesn't really fix the root issue (https://github.com/dotnet/runtime/issues/52977), it just reads the values and throws them away.

I think @SamMonoRT wanted @ivanpovazan to work on a proper fix.

We do care about this on non-windows platforms, by the way.  we want to support `[UnmanagedCallersOnly (CallConvs = new Type[] {CallConvCdecl, CallConvSuppressGCTransition})]`, for example.

I guess this is ok if it unblocks https://github.com/dotnet/runtime/issues/57361 if we want to do that quickly for now

Reply:
This should be ok for a quick fix to unblock 57361. @ivanpovazan will work on a proper fix and maybe change this temporary fix during that time. He'll start investigating this sometime this week.